File: frontend/lib/domain/entities/attendance.dart
class AttendanceSheetEntry {
  const AttendanceSheetEntry({
    required this.studentId,
    required this.studentName,
    this.status,
    this.observations,
  });

  final String studentId;
  final String studentName;

  /// Um de: PRESENT, ABSENT, LATE ou null (aluno ainda sem registro).
  final String? status;
  final String? observations;

  AttendanceSheetEntry copyWith({String? status, String? observations}) => AttendanceSheetEntry(
        studentId: studentId,
        studentName: studentName,
        status: status ?? this.status,
        observations: observations ?? this.observations,
      );
}

class AttendanceSheet {
  const AttendanceSheet({
    required this.lessonId,
    required this.classId,
    required this.attendanceCompleted,
    required this.students,
  });

  final String lessonId;
  final String classId;
  final bool attendanceCompleted;
  final List<AttendanceSheetEntry> students;
}

File: backend/src/modules/attendance/domain/attendance.ts
export type AttendanceStatus = 'PRESENT' | 'ABSENT' | 'LATE';

export type AttendanceRecord = {
  id: string;
  teacherId: string;
  lessonId: string;
  studentId: string;
  status: AttendanceStatus;
  observations: string | null;
  createdAt: Date;
  updatedAt: Date;
};

export type AttendanceSheetEntry = {
  studentId: string;
  studentName: string;
  status: AttendanceStatus | null;
  observations: string | null;
};

export type AttendanceSheet = {
  lessonId: string;
  classId: string;
  attendanceCompleted: boolean;
  students: AttendanceSheetEntry[];
};
